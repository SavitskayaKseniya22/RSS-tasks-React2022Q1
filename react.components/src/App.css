.App {
  max-width: 100vw;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}

* {
  box-sizing: border-box;
}

ul {
  list-style: none;
  margin: 0;
}
