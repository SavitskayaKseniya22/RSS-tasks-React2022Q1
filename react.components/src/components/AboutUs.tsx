import React from 'react';

export class AboutUs extends React.Component {
  constructor(props: string) {
    super(props);
  }
  render() {
    return (
      <div>
        <h1>about us</h1>
      </div>
    );
  }
}
