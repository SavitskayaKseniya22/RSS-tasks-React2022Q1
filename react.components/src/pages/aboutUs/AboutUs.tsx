import React from 'react';

export class AboutUs extends React.Component {
  constructor(props: string) {
    super(props);
  }
  render() {
    return (
      <div data-testid="about-us">
        <h1>about us</h1>
      </div>
    );
  }
}
