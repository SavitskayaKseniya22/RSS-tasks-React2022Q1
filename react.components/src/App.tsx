import React from 'react';
import './App.css';
import { MainPage } from './components/MainPage';

function App() {
  return (
    <div className="App">
      <header className="App-header"></header>
      <MainPage />
    </div>
  );
}

export default App;
