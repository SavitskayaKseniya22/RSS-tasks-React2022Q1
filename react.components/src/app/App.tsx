import { Navigate, Outlet, Route, Routes } from 'react-router-dom';
import './App.css';
import { MainPage } from '../pages/mainPage/MainPage';
import { MyAds } from '../pages/myAds/MyAds';
import { AboutUs } from '../pages/aboutUs/AboutUs';
import { Header } from '../components/header/Header';
import { Footer } from '../components/footer/Footer';
import { Dispatch, useReducer } from 'react';
import { GlobalTypes, ReducerTypes } from '../interfaces';
import React from 'react';
import { SearchItemDetails } from '../components/searchItemDetails/SearchItemDetails';

export const initialValues: GlobalTypes = {
  value: window.localStorage.getItem('searchValue') || '',
  response: [],
  isDownloading: false,
  isSearchOver: false,
  isError: false,
  activeCard: undefined,
  sort: 'latest',
  itemsPerPage: '20',
  pageNumber: '1',
  maxPageNumber: 10,
  isMounted: true,
  savedCards: [],
};

export const ContextApp = React.createContext<{
  state: GlobalTypes;
  dispatch: Dispatch<ReducerTypes>;
}>({
  state: initialValues,
  dispatch: () => null,
});

export function reducer(state: GlobalTypes, action: ReducerTypes): GlobalTypes {
  switch (action.type) {
    case 'handleDownload':
      return {
        ...state,
        response: action.payload.response,
        isDownloading: action.payload.isDownloading,
        isSearchOver: !action.payload.isDownloading,
        isError: action.payload.isError || false,
      };

    case 'toggleCard':
      return { ...state, activeCard: action.payload.activeCard };

    case 'handleSearchForm':
      return {
        ...state,
        sort: action.payload.sort,
        itemsPerPage: action.payload.itemsPerPage,
        pageNumber: action.payload.pageNumber,
        maxPageNumber: action.payload.maxPageNumber,
        value: action.payload.value,
        isMounted: action.payload.isMounted,
      };

    case 'handleSavedCards':
      return {
        ...state,
        savedCards: action.payload.savedCards,
      };
    case 'handleAddsForm':
      return {
        ...state,
        addsFormValues: action.payload.addsFormValues,
      };

    default:
      return state;
  }
}
function App() {
  const [state, dispatch] = useReducer(reducer, initialValues);

  return (
    <div className="App" data-testid="app">
      <ContextApp.Provider value={{ state, dispatch }}>
        <Header />
        <Footer />
        <Outlet />
        <Routes>
          <Route path="/my-ads" element={<MyAds />} />
          <Route path="/" element={<MainPage />} />
          <Route path="/about-us" element={<AboutUs />} />
          <Route
            path="/card"
            element={state.activeCard ? <SearchItemDetails /> : <Navigate to="/" />}
          />

          <Route
            path="/*"
            element={
              <div className="empty-page">
                <span>404</span>
              </div>
            }
          />
        </Routes>
      </ContextApp.Provider>
    </div>
  );
}

export default App;
