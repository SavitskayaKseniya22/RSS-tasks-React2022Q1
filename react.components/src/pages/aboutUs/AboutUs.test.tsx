import App from '../../App';
import { AboutUs } from './AboutUs';
import { MemoryRouter } from 'react-router-dom';
import { render, screen } from '@testing-library/react';

describe('AboutUs test', () => {
  test('check AboutUs page appearance', () => {
    render(<AboutUs />);
    const aboutUsPage = screen.getByTestId('about-us');
    expect(aboutUsPage).toMatchSnapshot();
  });

  test('check AboutUs page appearance with link', () => {
    render(
      <MemoryRouter initialEntries={['/about-us']}>
        <App />
      </MemoryRouter>
    );
    expect(screen.getByTestId('about-us')).toBeInTheDocument();
  });
});
