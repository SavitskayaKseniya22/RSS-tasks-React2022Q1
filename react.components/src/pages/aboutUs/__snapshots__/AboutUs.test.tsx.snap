// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutUs test check AboutUs page appearance 1`] = `
<div
  data-testid="about-us"
>
  <h1>
    about us
  </h1>
</div>
`;
